sway-lib-std/src/tx.sw

```diff
@@ -139,24 +141,30 @@ pub fn tx_script_start_offset() -> u32 {
 // Script
 ////////////////////////////////////////
 
+/// Get the transaction script data start offset.
+pub fn tx_script_data_start_offset() -> u32 {
+    asm(r1, r2: TX_SCRIPT_START_OFFSET, r3: TX_SCRIPT_LENGTH_OFFSET) {
+        lw r3 r3 i0;
+        add r1 r2 r3;
+        r1: u32
+    }
+}
+
 /// Get the script data, typed. Unsafe.
-// pub fn get_script_data<T>() -> T {
-//     // TODO some safety checks on the input data? We are going to assume it is the right type for now.
-//     // TODO test this before uncommenting.
-//     asm(script_data_len, to_return, script_data_ptr, script_len, script_len_ptr: 10280, script_data_len_ptr: 10288) {
-//         lw script_len script_len_ptr i0;
-//         lw script_data_len script_data_len_ptr i0;
-//         // get the start of the script data
-//         // script_len + script_start
-//         add script_data_ptr script_len is;
-//         // allocate memory to copy script data into
-//         aloc script_data_len;
-//         move to_return sp;
-//         // copy script data into above buffer
-//         mcp to_return script_data_ptr script_data_len;
-//         to_return: T
-//     }
-// }
+pub fn get_script_data<T>() -> T {
+    // TODO some safety checks on the input data? We are going to assume it is the right type for now.
+    let ptr = tx_script_data_start_offset();
+    if is_reference_type::<T>() {
```

Comment:
Can we replace this `if/else` with a call to `mem::read()`? 😄

Reply:
Yes!

The same can also be done for: https://github.com/FuelLabs/sway/pull/1984

cc @camsjams

Reply:
@mohammadfawaz created an issue: https://github.com/FuelLabs/sway/issues/2015